tuvali/tuvali-index.tsx

```diff
@@ -0,0 +1,78 @@
+import {NativeEventEmitter, NativeModules, Platform} from 'react-native';
+import type {Verifier, VersionModule, Wallet} from './types/interface';
+import {tuvaliVersion} from '../screens/tuvaliVersion';
+import {EventTypes, VerificationStatus} from './types/events';
+
+const LINKING_ERROR =
+  `The package '@mosip/tuvali' doesn't seem to be linked. Make sure: \n\n` +
+  Platform.select({ios: "- You have run 'pod install'\n", default: ''}) +
+  '- You rebuilt the app after installing the package\n' +
+  '- You are not using Expo Go\n';
+
+const VERIFIER_NOT_IMPLEMENTATED_ERROR = `Verifier is not yet implemented on IOS. Please remove Verifier usage on IOS Platform`;
+const isIOS = Platform.OS === 'ios';
+
+const versionModule: VersionModule = NativeModules.VersionModule
+  ? NativeModules.VersionModule
+  : new Proxy(
+      {},
+      {
+        get() {
+          throw new Error(LINKING_ERROR);
+        },
+      },
+    );
+
+const wallet: Wallet = NativeModules.WalletModule
+  ? NativeModules.WalletModule
+  : new Proxy(
+      {},
+      {
+        get() {
+          throw new Error(LINKING_ERROR);
+        },
+      },
+    );
+
+// TODO: Use Actual Verifier module on IOS once Verifier is implemented
+let verifier: Verifier = NativeModules.VerifierModule
+  ? NativeModules.VerifierModule
+  : versionModule.setTuvaliVersion(tuvaliVersion);
+
+if (!isIOS) {
+  setupModule(verifier);
+}
```

Comment:
refactor the code to verifier instance, and setup verifier module
let verifier: Verifier = NativeModules.VerifierModule

```
if (!isIOS) {
	if (verifier)
		setupModule(verifier);
	else {
	// VERIFIER_NOT_IMPLEMENTATED_ERROR
	}
}

versionModule.setTuvaliVersion(tuvaliVersion);
```